docs/src/architecture/domain-model.md

```diff
@@ -0,0 +1,121 @@
+  <!-- TODO: add somehow "used to abtracts the hardware. But: on a microcontroler usually less abtraction, more hw dep on the domain app, and even no OS" -->
+* **Sensors** and **Actuators** enable the **device** to observe and control physical behaviour on the shopfloor or device's environment.
+  * can be integrated in the **device's** hardware,
+    or connected to the **device** via some **Fieldbus** (e.g. Modbus, CANopen, PROFINET, EtherCAT, ...) or
+    some **Local Interface** as USB, UART, SPI, I2C, ...
+  * can be simple peripherals as a light barrier, or a sensor for temperature or pressure;
+    or complex equipments as robotic arms or even other **devices**
+* **Inputs / Outputs** are the communication channels between the **Domain Application** and **Sensors** and **Actuators**
+  * drivers (as part of the **OS / Libs / Runtime** and/or the **Domain Application**) do expose all data from
+    **Sensors** and **Actuators** to the **device** as inputs or outputs
+  * also the **Domain Application** can expose data as input or output (e.g. own _signals_ or _states_)
+
+## thin-edge data concepts
+
+**thin-edge** provides different devices in a standardized representation to the cloud. Therefore **thin-edge** provides specialized data concepts.
+
+The data concepts below belong to **inputs** and **outputs** available on a device. Thereby each **input** and **output** is refered as a **data point**.
+
+  * **Measurements**:
+    * contain numeric data produced by sensors (like temperature readings) or calculated data based on information from the **control application**.
+    * a **measurements** consists of one or more numeric **data points** and optionally meta information as names or units
+  * **Variables**:
+    * contain numeric data provided and used by the device, and can be sent to the device (as e.g. "set points")
+    * a **variable** consists of one or more numeric **data points** and optionally meta information as names or units
+  * **Events**:
+    * contain other real-time information from the sensor network, such as the triggering of a door sensor
+    * an **event** is triggered by a **data point** (e.g. when it changes to a specific value as `0` or `1`), and contains an optional message text.
+  * **Alarms**:
+    * similar to Events, but the user or operator of the system has to take action to resolve the alarm
+    * an **alarm** is triggered by a **data point** (e.g. when it changes to a specific value as `0` or `1`), and contains an optional alarm text.
+
+## thin-edge device concept
+
+**thin-edge** facilitates IoT functionality to the device it is running on, as well as to devices that are connected to that device.
+All devices are categorized as below:
+  * the **main-device** is the device thin-edge is running on
+  * all devices connected to the **main-device** are referred as **external child-devices**
+  * each device has a unique **device-id**
```

Comment:
Here the cloud aspects are missing, e.g.:
  - How is the main-device represented in the cloud?
  - How is an external child-device represented in the cloud?

I am going to add those aspects.

Reply:
Cloud aspect is now added to section "thin-edge device concept".